### PR TITLE
Infinite list

### PR DESCRIPTION
**Pull request by filiph**
Adding a new sample:

screencapture-1589399605839 2020-05-13 12_56_13

I've been asked about this a lot in the past few months.

### CODE REVIEW

**Comment by johnpryan** [comment]
We can also make this a web demo on the [samples page](https://flutter.github.io/samples/#?platform=web)

**Comment by filiph** [comment]
@johnpryan Sure! That would be great.

I'm confused why the Travis CI check hasn't started yet.

File: infinite_list/analysis_options.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+include: package:pedantic/analysis_options.yaml
+
+analyzer:
+  strong-mode:
+    implicit-casts: false
+    implicit-dynamic: false
```
Comment (domesticmouse) [inline infinite_list/analysis_options.yaml:6]: We have some additional tests that we like our samples to pass:

```yaml
linter:
  rules:
    - avoid_types_on_closure_parameters
    - avoid_void_async
    - await_only_futures
    - camel_case_types
    - cancel_subscriptions
    - close_sinks
    - constant_identifier_names
    - control_flow_in_finally
    - directives_ordering
    - empty_statements
    - hash_and_equals
    - implementation_imports
    - non_constant_identifier_names
    - package_api_docs
    - package_names
    - package_prefixed_library_names
    - test_types_in_equals
    - throw_in_finally
    - unnecessary_brace_in_string_interps
    - unnecessary_getters_setters
    - unnecessary_new
    - unnecessary_statements
```

**Comment by domesticmouse** [comment]
btw, @filiph your PR is [green on travis](https://travis-ci.org/github/flutter/samples/builds/686738960)

File: infinite_list/analysis_options.yaml (hunk 1 above)
Comment (filiph) [reply infinite_list/analysis_options.yaml:6]: Ack, adding.

**Comment by RedBrogdon** [comment]
@filiph, I won't have time until tomorrow, but I'd love to give this a review before you land it.

**Comment by filiph** [comment]
No hurry, I'll wait!

**Comment by filiph** [comment]
Thanks! I'm glad for every additional pair of eyes.

The CI is green, I will squash and merge now. A heads up: one of the "allowed failure" tests failed: https://travis-ci.org/github/flutter/samples/jobs/687657522. It's something with `dynamic` in `jsonexample/`. Fails on the Flutter beta channel (1.18.0-11.1.pre).

**Pull request merged**